src/openforms/js/components/admin/form_design/registrations/CatalogiSelect.js

```diff
@@ -0,0 +1,77 @@
+import PropTypes from 'prop-types';
+import React, {useEffect} from 'react';
+
+import ReactSelect from 'components/admin/forms/ReactSelect';
+
+const CatalogiSelect = ({
+  name,
+  htmlId,
+  availableCatalogiState,
+  catalogusDomein,
+  catalogusRsin,
+  onChange,
+  isClearable = true,
+}) => {
+  const {loading, availableCatalogi, error} = availableCatalogiState;
+
+  const onCatalogChange = event => {
+    const {value} = event.target;
+    if (value === '') {
+      onChange({catalogusDomein: '', catalogusRsin: ''});
+    } else {
+      const splitValue = value.split('-');
+      onChange({catalogusDomein: splitValue[0], catalogusRsin: splitValue[1]});
+    }
+  };
+
+  const options =
+    loading || error
+      ? []
+      : availableCatalogi.map(({domein, rsin}) => {
+          return {value: `${domein}-${rsin}`, label: `${domein} (RSIN: ${rsin})`};
+        });
```

Comment:
Hum I don't see how this can be done without complicating things for simple usages, e.g.:
https://github.com/open-formulieren/open-forms/blob/8dccef94dd72179ee1cec066ddb732e1c4e5491e/src/openforms/js/components/admin/form_design/registrations/objectsapi/V2ConfigFields.js#L144-L150

Especially the logic in `extractSelectedValue` to build the react-select `value` prop. I also wanted to stay as close as possible to the API we defined in the formio-builder

Reply:
the idea of react-select is that you can pass a list of options (where each item is some object) and then pass one of the items of that list as `value` too, and react-select would do the comparison for you :thinking: but I think we're in a terrible state at the moment where no decision works for all the use cases because we have so many variations :(